File: esthesis-platform-backend/esthesis-platform-backend-server/src/main/java/esthesis/platform/backend/server/dto/DashboardWidgetDTO.java
package esthesis.platform.backend.server.dto;

import esthesis.platform.backend.common.dto.BaseDTO;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
public class DashboardWidgetDTO extends BaseDTO {

  private String type;
  private int gridCols;
  private int gridRows;
  private int gridX;
  private int gridY;
  private String configuration;
  private long dashboard;
  private int updateEvery;
}

File: ui/src/app/dashboard/dto/dashboard-widget-dto.ts
import {BaseDto} from "../../dto/base-dto";

export interface DashboardWidgetDto extends BaseDto {
  type: string;
  gridCols: number;
  gridRows: number;
  gridX: number;
  gridY: number;
  configuration: any;
  dashboard: number;
  updateEvery: number;
}
